lib/src/puppetdb-cli.cc

```diff
@@ -116,19 +117,32 @@ size_t write_body(char *ptr, size_t size, size_t nmemb, void *userdata){
     return size * nmemb;
 }
 
+string
+convert_query_to_post_data(const string& query_str) {
+    // If this is PQL then we need to wrap the query in double-quotes, otherwise
+    // the query is AST and we leave it alone
+    string query_str_copy { query_str };
+    boost::trim(query_str_copy);
+    const string query = (boost::starts_with(query_str_copy, "[")) ?
+            query_str : "\""+ query_str +"\"";
```

Comment:
I don't think it can be this simple. I think it needs to be parsed/emitted JSON. It breaks with the following invocation:

```
build/bin/puppet-query -l debug 'nodes{certname ~ ".*"}'
```